core/Pos/Binary/Merkle.hs

```diff
@@ -4,16 +4,16 @@ module Pos.Binary.Merkle () where
 
 import           Universum
 
-import           Pos.Binary.Class   (Bi (..), Raw, label, putField)
+import           Pos.Binary.Class   (Bi (..), Raw)
 import           Pos.Crypto.Hashing (Hash)
 import           Pos.Merkle         (MerkleRoot (..), MerkleTree (..), mkMerkleTree)
 
 -- This instance is both faster and more space-efficient (as confirmed by a
 -- benchmark). Hashing turns out to be faster than decoding extra data.
 instance (Bi a, Bi (Hash Raw)) => Bi (MerkleTree a) where
-    sizeNPut = putField toList
-    get = label "MerkleTree" $ mkMerkleTree <$> get
+  encode = encode . toList
+  decode = mkMerkleTree <$> decode
 
 instance (Bi a, Bi (Hash Raw)) => Bi (MerkleRoot a) where
-    sizeNPut = putField getMerkleRoot
-    get = label "MerkleRoot" $ MerkleRoot <$> get
+  encode = encode . getMerkleRoot
+  decode = MerkleRoot <$> decode
```

Comment:
Please be more careful to the number of spaces. We use 4 spaces. I see that in many places you changed number of places from 4 to 2, but in other places you preserved 4 spaces.
It's not critical and I don't insist on fixing it, but personally I don't like how it looks when I read it